File: Data Structures & Algorithms/max-water-container/submission-2.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l = 0, r = heights.length - 1;
        let max = 0;
        while (l < r) {
            // calculate the amount
            let width = r - l;
            let height = Math.min(heights[l], heights[r]);
            max = Math.max(max, (width * height));

            // shift whichever height is shorter
            if (heights[l] > heights[r]) {
                r--;
            } else {
                l++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        low = prices[0]
        max_profit = 0

        for p in prices:
            max_profit = max(max_profit, p - low)
            low = min(low, p)
        
        return max_profit
        